src/hip_hcc.cpp

```diff
@@ -1409,9 +1410,38 @@ void ihipInit()
     tprintf(DB_SYNC, "pid=%u %-30s g_numLogicalThreads=%u\n", getpid(), "<ihipInit>", g_numLogicalThreads);
 }
 
+hipError_t ihipStreamSynchronize(hipStream_t stream)
+{
+    hipError_t e = hipSuccess;
 
+    if (stream == hipStreamNull) {
+        ihipCtx_t *ctx = ihipGetTlsDefaultCtx();
+        ctx->locked_syncDefaultStream(true/*waitOnSelf*/, true/*syncToHost*/);
+    } else {
+        // note this does not synchornize with the NULL stream:
+        stream->locked_wait();
+        e = hipSuccess;
+    }
 
+    return e;
+}
 
+void ihipStreamCallbackHandler(ihipStreamCallback_t *cb)
+{
+    hipError_t e = hipSuccess;
+
+    // Notify hipStreamAddCallback that callback handler thread is active
+    std::lock_guard<std::mutex> guard(cb->_mtx);
+    cb->_ready = true;
+
+    // Synchronize stream
+    tprintf(DB_SYNC, "ihipStreamCallbackHandler wait on stream %s\n", ToString(cb->_stream).c_str());
+    e = ihipStreamSynchronize(cb->_stream);
```

Comment:
e might be initialized here.

Reply:
I am not sure what you mean by initialized here. Are you saying that we shouldn't initialize e to hipSuccess at the beginning of the function? It is a common practice in the rest of the code.

Reply:
Yes, it should be initialized where first assigned, as it is not used anywhere before. It just eliminates 1 unneeded assignment (performance).

Reply:
Firstly it is just that it is good coding practice to initialize the variable. Also I think the compiler will optimize this anyways.